src/lib/components/form/Form.component.tsx

```diff
@@ -0,0 +1,346 @@
+import {
+  Children,
+  createContext,
+  HTMLProps,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import styled from 'styled-components';
+import { spacing, Stack, Wrap } from '../../spacing';
+import { convertRemToPixels, getTheme } from '../../utils';
+import { Box } from '../box/Box';
+import { Icon, IconName } from '../icon/Icon.component';
+import { IconHelp } from '../IconHelper';
+import { ScrollbarWrapper } from '../scrollbarwrapper/ScrollbarWrapper.component';
+import { Text } from '../text/Text.component';
+
+const DESCRIPTION_PREFIX = 'describe-';
+
+type FormProps = HTMLProps<HTMLFormElement> & {
+  layout: 'page' | 'tab';
+  children: ReactNode | ReactNode[];
+  requireMode?: 'all' | 'partial';
+  leftActions?: ReactNode;
+  rightActions?: ReactNode;
+  title?: string;
+  banner?: ReactNode;
+};
+
+const PageFormWrapper = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  height: 100%;
+`;
+
+const BasicPageLayout = styled.div`
+  margin: 0 auto;
+  width: 45rem;
+`;
+
+const FixedHeader = styled(BasicPageLayout)`
+  border-bottom: 1px solid ${(props) => getTheme(props).border};
+`;
+
+const FixedFooter = styled(BasicPageLayout)`
+  border-top: 1px solid ${(props) => getTheme(props).border};
+`;
+
+const PaddedContent = styled.div`
+  padding: ${spacing.f16} 0 ${spacing.f16} ${spacing.f16};
+`;
+const PaddedForHeaderAndFooterContent = styled.div`
+  padding: ${spacing.f16};
+`;
+
+const ScrollArea = styled(BasicPageLayout)`
+  flex-grow: 1;
+  align-self: stretch;
+  overflow-y: auto;
+`;
+
+type ContentProps = {
+  helper: string;
+  error: string;
+};
+
+type FormGroupProps = {
+  direction: 'vertical' | 'horizontal';
+  label: string;
+  id: string;
+  content: ReactElement<ContentProps>;
+  labelHelpTooltip?: string;
+  help?: string;
+  error?: string;
+  required?: boolean;
+  helpErrorPosition?: 'right' | 'bottom';
+  disabled?: boolean;
+};
+
+const LabelContext = createContext({
+  maxLabelWidth: 0,
+  setMaxLabelWidth: (setter: (value: number) => number) => {},
+});
+
+const RequireModeContext = createContext<'all' | 'partial'>('partial');
+
+const FormGroup = ({
+  direction,
+  label,
+  id,
+  labelHelpTooltip,
+  content,
+  help,
+  error,
+  required,
+  helpErrorPosition,
+  disabled,
+}: FormGroupProps) => {
+  const { maxLabelWidth, setMaxLabelWidth } = useContext(LabelContext);
```

Comment:
we might want to add a label width enforced at the section level and stored within this context.